src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -989,7 +989,7 @@ A characterization: `simple_graph.is_acyclic_iff`.-/
 def is_acyclic : Prop := ∀ (v : V) (c : G.walk v v), ¬c.is_cycle
 
 /-- A *tree* is a connected acyclic graph. -/
-def is_tree : Prop := G.connected ∧ G.is_acyclic
+def is_tree : Prop := G.connected ∧ G.is_acyclic ∧ nonempty V
```

Comment:
With 3 fields I'd turn it into a `structure` with meaningful projection names.
```suggestion
@[protect_proj] structure is_tree : Prop :=
(connected : G.connected)
(is_acyclic : G.is_acyclic)
(nonempty : nonempty V)
```

Then you can use `h.connected`, `h.is_acyclic`, and `h.nonempty`.